fix: TypeScript-Fehler in LevelImageDisplay und ThemeContext

## Problem

`npx tsc --noEmit` meldet 3 Fehler in zwei Dateien. Diese sind pre-existierend und nicht mit einem bestimmten Feature verknüpft.

## Fehler

### `components/LevelImageDisplay.tsx:601`
```
error TS18046: 'svgElement.props' is of type 'unknown'.
```
`svgElement.props` muss gecastet oder typisiert werden.

### `services/ThemeContext.tsx:190` und `:207`
```
error TS2345: Argument of type 'ColorSchemeName' is not assignable to parameter of type 'SetStateAction<"light" | "dark">'.
  Type '"unspecified"' is not assignable to type 'SetStateAction<"light" | "dark">'.
```
`ColorSchemeName` von React Native kann `"unspecified"` sein, aber der State ist auf `"light" | "dark"` eingeschränkt. Der `"unspecified"`-Fall muss vor dem `setState`-Aufruf gefiltert werden.

## Fix-Hinweise

- `ThemeContext.tsx`: Vor `setTheme(colorScheme)` prüfen: `if (colorScheme === 'unspecified' || !colorScheme) return;`
- `LevelImageDisplay.tsx`: `svgElement.props` als `React.ComponentProps<...>` oder `any` casten (mit Kommentar)

## Priorität

Niedrig — kein Laufzeit-Problem bekannt, aber sollte sauber sein vor dem nächsten Play-Store-Release.

--- a/app/game.tsx
+++ b/app/game.tsx
@@ -284,7 +284,7 @@ export default function GameScreen() {
     return (
       <ScrollView
         style={[styles.phaseContainer, { padding: 0 }]}
-        contentContainerStyle={[styles.resultContent, { paddingBottom: Math.max(insets.bottom, Spacing.md) }]}
+        contentContainerStyle={[styles.resultContent, { paddingBottom: Spacing.md }]}
         showsVerticalScrollIndicator={false}
       >
         <Text style={styles.phaseTitle}>{t('game.result.title')}</Text>
@@ -380,7 +380,7 @@ export default function GameScreen() {
   };
 
   return (
-    <View style={styles.container}>
+    <View style={[styles.container, { paddingBottom: insets.bottom }]}>
       {/* Header */}
       <View style={[styles.header, { paddingVertical: layout.headerPaddingVertical, paddingHorizontal: layout.headerPaddingHorizontal }]}>
         <TouchableOpacity

--- a/app/game.tsx
+++ b/app/game.tsx
@@ -284,7 +284,7 @@ export default function GameScreen() {
     return (
       <ScrollView
         style={[styles.phaseContainer, { padding: 0 }]}
-        contentContainerStyle={[styles.resultContent, { paddingBottom: Spacing.md }]}
+        contentContainerStyle={styles.resultContent}
         showsVerticalScrollIndicator={false}
       >
         <Text style={styles.phaseTitle}>{t('game.result.title')}</Text>

--- a/components/LevelImageDisplay.tsx
+++ b/components/LevelImageDisplay.tsx
@@ -598,7 +598,7 @@ export default function LevelImageDisplay({ image, size = 300, revealStep }: Pro
 
   // Progressive reveal: only show children up to revealStep
   if (revealStep !== undefined) {
-    const children = React.Children.toArray(svgElement.props.children);
+    const children = React.Children.toArray((svgElement.props as { children?: React.ReactNode }).children);
     const visibleChildren = children.slice(0, revealStep + 1);
     const cloned = React.cloneElement(svgElement, {}, ...visibleChildren);
     return (

--- a/components/LevelImageDisplay.tsx
+++ b/components/LevelImageDisplay.tsx
@@ -598,7 +598,7 @@ export default function LevelImageDisplay({ image, size = 300, revealStep }: Pro
 
   // Progressive reveal: only show children up to revealStep
   if (revealStep !== undefined) {
-    const children = React.Children.toArray((svgElement.props as { children?: React.ReactNode }).children);
+    const children = React.Children.toArray((svgElement.props as React.ComponentProps<typeof Svg>).children);
     const visibleChildren = children.slice(0, revealStep + 1);
     const cloned = React.cloneElement(svgElement, {}, ...visibleChildren);
     return (

--- a/services/ThemeContext.tsx
+++ b/services/ThemeContext.tsx
@@ -187,7 +187,8 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
     let isMounted = true;
 
     // Set initial system theme after mount (client-side only)
-    setSystemTheme(Appearance.getColorScheme() || 'light');
+    const initialScheme = Appearance.getColorScheme();
+    setSystemTheme(initialScheme === 'dark' ? 'dark' : 'light');
 
     const initTheme = async () => {
       try {
@@ -204,7 +205,7 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
     // Listen to system theme changes
     const subscription = Appearance.addChangeListener(({ colorScheme }) => {
       if (isMounted) {
-        setSystemTheme(colorScheme || 'light');
+        setSystemTheme(colorScheme === 'dark' ? 'dark' : 'light');
       }
     });
 

--- a/services/ThemeContext.tsx
+++ b/services/ThemeContext.tsx
@@ -9,6 +9,11 @@ import storageManager from './StorageManager';
 
 export type Theme = 'light' | 'dark' | 'system';
 
+/** Maps ColorSchemeName (which can be null or "unspecified") to a concrete light/dark value. */
+function normalizeColorScheme(scheme: ReturnType<typeof Appearance.getColorScheme>): 'light' | 'dark' {
+  return scheme === 'dark' ? 'dark' : 'light';
+}
+
 export interface ThemeColors {
   // Primary
   primary: string;
@@ -187,8 +192,7 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
     let isMounted = true;
 
     // Set initial system theme after mount (client-side only)
-    const initialScheme = Appearance.getColorScheme();
-    setSystemTheme(initialScheme === 'dark' ? 'dark' : 'light');
+    setSystemTheme(normalizeColorScheme(Appearance.getColorScheme()));
 
     const initTheme = async () => {
       try {
@@ -205,7 +209,7 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
     // Listen to system theme changes
     const subscription = Appearance.addChangeListener(({ colorScheme }) => {
       if (isMounted) {
-        setSystemTheme(colorScheme === 'dark' ? 'dark' : 'light');
+        setSystemTheme(normalizeColorScheme(colorScheme));
       }
     });
 

